core/serialization/deserializer/montage-reviver.js

```diff
@@ -502,7 +502,14 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
                 object = value && "prototype" in value ?
                     Object.create(module.montageObject) : module.montageObject;
                 context.setObjectLabel(object, label);
-                return this.instantiateMJSONObject(value, object, objectName, context, label);
+
+                if(label === "root") {
+                    return this.instantiateMontageObject(value, object, objectName, context, label);
+                }
+                else {
+                    return this.instantiateMJSONObject(value, object, objectName, context, label);
```

Comment:
An object doesn't receive `deserializedSelf` if it's the root? Isn't that an issue that should be fixed in `instantiateMJSONObject`?

Reply:
Will look more into it

Reply:
This has been taken care of